packages/shared/src/permissions/permissionsClass.ts

```diff
@@ -55,6 +59,46 @@ export class Permissions {
     );
   };
 
+  // This is a helper method to use on the frontend to determine whether or not a user can create Fact Tables in current project
+  public canViewCreateFactTableModal = (project?: string): boolean => {
+    return this.checkProjectFilterPermission(
+      {
+        projects: project ? [project] : [],
+      },
+      "manageFactTables"
+    );
+  };
+
+  // This is a helper method to use on the frontend to determine whether or not a user can edit a Fact Table
+  public canViewEditFactTableModal = (
+    factTable: Pick<FactTableInterface, "projects">
+  ): boolean => {
+    return this.checkProjectFilterPermission(factTable, "manageFactTables");
+  };
```

Comment:
Open to feedback on whether or not others think having these two separate helpers is worth it. I ultimately opted for both given the signatures are different. While having two separate helpers adds weight to the class, I think it'll improve the developer experience.